packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1954,6 +1954,7 @@ def setup(self) -> None:
         """Set up the behaviour."""
         self._schedule_round(self.initial_round_cls)
         if self.is_termination_set:
+            self.background_round_cls = cast(AppState, self.background_round_cls)
             self._background_round = self.background_round_cls(
```

Comment:
I know it's very annoying this casting for mypy, makes code messy, next time consider making it a property so you don't have to keep casting. A property that raises if not `self.is_termination_set` or so. Just a suggestion. I also kinda dislike the fact that 3 class attribute are needed for this, and you can guess why I'm sure but anyway: we use too many class attributes and they are the devil because state caries over more easily than when you're using an instance attribute.